demos/chat_with_pdf.py

```diff
@@ -318,14 +346,6 @@ def update_llm(self):
 
         print(f"DEBUG: Updated LLM type: {type(self.cached_llm)}")
 
-    def update_chain(self):
```

Comment:
@bsbodden self.qa_chain didn't appear to be used anywhere so I think this was dead code that we can clean out

Reply:
Indeed, I left it behind while refactoring. Thanks